src/components/tooltip/tooltip.js

```diff
@@ -10,6 +35,159 @@ import {h, Component} from 'preact';
  * @extends {Component}
  */
 class Tooltip extends Component {
+  state: Object;
+  _hoverTimeout: ?number;
+  textElement: HTMLSpanElement;
+  nonValidTypes: string[] = [];
+
+  /**
+   * default component props
+   * @type {Object}
+   * @memberof Tooltip
+   */
+  static defaultProps = {
+    type: ToolTipType.Top,
+    isMultiline: true,
+    maxWidth: '240px'
+  };
+
+  /**
+   * @constructor
+   * @param {*} props props
+   * @memberof Tooltip
+   */
+  constructor(props: Object) {
+    super(props);
+    this.setState({showTooltip: false, marginTop: 0, marginLeft: 0});
+  }
+
+  /**
+   * gets the tooltip position based on component prop
+   *
+   * @returns {void}
+   * @memberof Tooltip
+   */
+  setMargins() {
```

Comment:
why don't you create css class and just set the correct class? I think it's too much style work on js.

Reply:
Yep :) done